packages/core/contracts/zodiac/OptimisticOracleModule.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.6;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "../oracle/implementation/Constants.sol";
+import "../oracle/interfaces/FinderInterface.sol";
+import "../oracle/interfaces/SkinnyOptimisticOracleInterface.sol";
+import "../oracle/interfaces/OracleAncillaryInterface.sol";
+
+contract OptimisticOracleModule is Module {
+    event OptimisticOracleModuleDeployed(address indexed owner, address indexed avatar, address target);
+
+    event TransactionsProposed(uint256 indexed proposalId, address indexed proposer, uint256 indexed proposalTime);
+
+    event TransactionExecuted(uint256 indexed proposalId, uint256 indexed transactionIndex);
+
+    event ProposalDeleted(uint256 indexed proposalId);
+
+    FinderInterface public finder;
+    SkinnyOptimisticOracleInterface optimisticOracle;
+
+    IERC20 public collateral;
+    uint256 public bond;
+    string public rules;
+    uint256 public liveness;
+    bytes32 public identifier = "ZODIAC";
+
+    struct Transaction {
+        address to;
+        uint256 value;
+        bytes data;
+    }
+
+    struct Proposal {
+        Transaction[] transactions;
+        uint256 requestTime;
+        bytes ancillaryData;
+        bool status;
+    }
+
+    Proposal[] public proposals;
+
+    /**
+     * @notice Construct Optimistic Oracle Module.
+     * @param _finder Finder address.
+     * @param _owner Address of the owner.
+     * @param _collateral Address of the ERC20 collateral used for bonds.
+     * @param _bond Bond required (must be at least as large as final fee for collateral type).
+     * @param _rules Reference to the rules for the Gnosis Safe (e.g., IPFS hash or URI).
+     */
+    constructor(
+        address _finder,
+        address _owner,
+        address _collateral,
+        uint256 _bond,
+        string memory _rules
+    ) {
+        bytes memory initializeParams = abi.encode(_finder, _owner, _collateral, _bond, _rules);
+        setUp(initializeParams);
+    }
+
+    function setUp(bytes memory initializeParams) public override initializer {
```

Comment:
Does this work this way to allow this contract to optionally be used as a library (delegatecalled)? Or is setUp done this way for a different reason? @auryn-macmillan may be the right person to answer this question.

I just want to make sure we understand the usage pattern so we can make informed decisions about how we work with storage.

Reply:
The `setUp` function allows you to deploy a module like this at the same time as the Gnosis Safe that will be the module's avatar.

https://gnosis.github.io/zodiac/docs/tutorial-build-a-module/create_module#factory-friendly